algolia/algoliasearch-sfcc-b2c
# refactor: improved attributes configuration

Refactor to help build the extensibility feature (#150):
- `colorVariations` is now a normal attribute
  - `generateVariantRecordsWithColorVariations()` is now renamed to `generateVariantRecords()`: it's a generic function that builds variant records based on the declared attributes, and doesn't add systematically the `colorVariations` attribute.
- Added a way to declare attributes that should go in the `variants` for the product-level records model: `variantAttribute`
  - The jobs use this fields to build the `variantAttributes` array instead of using the default list
  - In `algoliaLocalizedProduct.js`, the `variants` handler accepts a second parameter, to pass that list

### How to test :test_tube: 
- Add `pricebooks` to the Additional Product Attributes
- Set the record model to "Base Product"
- Run the `AlgoliaProductIndex_v2` job.

Indexed records should contains the `pricebooks` fields inside the `variants` array (since it's declared with `variantAttribute: true`).
